### PR TITLE
Remove machines for k8s

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
- Remove the models machines tab on k8s
- Hide the Machine switcher on application view on kubernetes
- Hide machine table on units view on k8s 
- Remove machine column from app view

## QA
- You will need K8s model so ask me for that or we can pair on the QA
- Check all the above items are addressed

## Details
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1721

### CODE REVIEW

**Comment by anthonydillon** [comment]
@hatched if you get any time could you commit how you would conditionally remove the machine row data and header from the application view. The condition is: `modelStatusData.info["provider-type"] !== "kubernetes"`

**Comment by hatched** [comment]
There you go @anthonydillon It's a bit of a hacky approach but it gets the job done for now. I'm ok with it because we'll be refactoring all this with the new modular table in the near future.

**Comment by anthonydillon** [comment]
Thanks Jeff. Haha I wrote it this way and thought it was too hacky. I was worried that if someone changed the table it would hide the wrong thing and we would never spot it. 

But I think this could work if I add IDs to the columns and loop over and remove the one with machine as a key.

File: src/pages/EntityDetails/App/App.tsx [hunk 1]
```diff
@@ -204,11 +216,13 @@ export default function App(): JSX.Element {
         </>
       </div>
       <div className="entity-details__main u-overflow--scroll">
-        <ButtonGroup
-          buttons={["units", "machines"]}
-          activeButton={tableView}
-          setActiveButton={setTableView}
-        />
+        {modelStatusData?.info?.["provider-type"] !== "kubernetes" && (
```
Comment (hatched) [inline src/pages/EntityDetails/App/App.tsx:219]: Can you replace this check with the `hideMachines` value from above?

File: public/config.js [hunk 2]
```diff
@@ -3,15 +3,15 @@
 // eslint-disable-next-line no-unused-vars
 var jujuDashboardConfig = {
   // API host to allow app to connect and retrieve models
-  baseControllerURL: "jimm.jujucharms.com",
+  baseControllerURL: "localhost:17070",
```
Comment (hatched) [inline public/config.js:6]: Oops :) Can you remove these changes plz.

**Review by hatched — CHANGES_REQUESTED** [review]
Minor changes requested. Nice improvement on my hack :)

File: public/config.js (hunk 2 above)
Comment (anthonydillon) [reply public/config.js]: Done

File: src/pages/EntityDetails/App/App.tsx (hunk 1 above)
Comment (anthonydillon) [reply src/pages/EntityDetails/App/App.tsx]: Good spot, done

**Review by hatched — APPROVED** [review]
Thanks for keeping up with this 👍

**Pull request merged**